File: leetcode/p00167/p00167.go
package p00167

// Time Complexity: O(n)
// Space Complexity: O(1)
func twoSum(numbers []int, target int) []int {
	l, r := 0, len(numbers)-1

	for l < r {
		sum := numbers[l] + numbers[r]
		if sum == target {
			return []int{l + 1, r + 1}
		}
		if sum > target {
			r--
		} else {
			l++
		}
	}

	return []int{-1, -1}
}

File: leetcode/p00167/p00167.py
from typing import List

class Solution:
    # Time Complexity: O(n)
    # Space Complexity: O(1)
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers)-1

        while l < r:
            num = numbers[l] + numbers[r]
            if num == target:
                return [l+1, r+1]
            if num > target:
                r -= 1
            else:
                l += 1
        
        return [-1, -1]